src/network.h

```diff
@@ -991,12 +1038,25 @@ reac_norm calculate_reaction_norm(const Net& net,
                                   const range& cue_range,
                                   const int& n_data_points)
 {
+    //reac_norm r_norm;
+    //r_norm.reserve(n_data_points);
+    //double step_size = (cue_range.m_end - cue_range.m_start)/n_data_points;
+    //for(double i = cue_range.m_start; i < cue_range.m_end; i += step_size)
+    //{   
+    //  r_norm.push_back({ i, output(net, std::vector<double>{i}).at(0) });
+    //}
+    //return r_norm;
+
     reac_norm r_norm;
     r_norm.reserve(n_data_points);
-    double step_size = (cue_range.m_end - cue_range.m_start)/n_data_points;
-    for(double i = cue_range.m_start; i < cue_range.m_end; i += step_size)
+    std::vector<double> input;
+    std::vector<double> output;
+    double step_size = (cue_range.m_end - cue_range.m_start) / n_data_points;
+    for (double i = cue_range.m_start; i < cue_range.m_end; i += step_size)
     {
-        r_norm.insert({i, output(net, std::vector<double>{i}).at(0)});
+      input.assign({ i });
+      output_ugly_but_fast(net, input, output);
+      r_norm.emplace_back(i, output[0]);
```

Comment:
```suggestion
      r_norm.emplace_back(i, input[0]);
```
If I understand correctly the 
```c++
output_ugly_but_fast(net, input, output);
```
function always swaps output for input,  even in the last iteration, so I think it is input that holds the true output values.